packages/concerto-core/lib/introspect/field.js

```diff
@@ -62,24 +62,32 @@ class Field extends Property {
         this.validator = null;
 
         switch (this.getType()) {
-        case 'Integer':
-        case 'Double':
-        case 'Long':
-            if (this.ast.validator) {
-                this.validator = new NumberValidator(
-                    this,
-                    this.ast.validator
-                );
-            }
-            break;
-        case 'String':
-            if (this.ast.validator) {
-                this.validator = new StringValidator(
-                    this,
-                    this.ast.validator
-                );
-            }
-            break;
+            case 'Integer':
+            case 'Double':
+            case 'Long':
+                if (this.ast.validator) {
+                    this.validator = new NumberValidator(
+                        this,
+                        this.ast.validator
+                    );
+                }
+                break;
+            case 'String':
+                if (this.ast.validator) {
+                    this.validator = new StringValidator(
+                        this,
+                        this.ast.validator
+                    );
+                }
+                break;
+            case 'Map':
```

Comment:
Careful .. this looks like 'Map' is a new type of primitive - probably not want we want? There is lots of code that switches over the primitive types.